src/python/grpcio/grpc/aio/_interceptor.py

```diff
@@ -133,7 +268,7 @@ async def intercept_unary_unary(
         ],
         client_call_details: ClientCallDetails,
         request: RequestType,
-    ) -> Union[UnaryUnaryCall, ResponseType]:
+    ) -> UnaryUnaryCall:
```

Comment:
We should globally disable this check in typeguard. If our tests don't have coverage for returning `ResponseType` that we claim is a valid return type, it's on us to add the missing coverage.